### ISSUE TITLE
Bad username or password during setup

### ISSUE THREAD

**Issue by Austinzveare** [boredazfcuk/docker-icloudpd#622]
I just set this up using the Unraid template but can't seem to get past entering my Apple ID password. I have it set to MFA.

`Traceback (most recent call last):
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/base.py", line 241, in _authenticate_with_token
    req = self.session.post(
          ^^^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/requests/sessions.py", line 637, in post
    return self.request("POST", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/session.py", line 167, in request
    self._raise_error(code or "Unknown", reason)
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/session.py", line 196, in _raise_error
    raise api_error
pyicloud_ipd.exceptions.PyiCloudAPIResponseException: Missing apple_id field (Missing apple_id field)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/cmdline.py", line 257, in main_aux
    api = PyiCloudService(
          ^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/base.py", line 157, in __init__
    self.authenticate()
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/base.py", line 221, in authenticate
    self._authenticate_with_token()
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/base.py", line 247, in _authenticate_with_token
    raise PyiCloudFailedLoginException(msg, error) from error
pyicloud_ipd.exceptions.PyiCloudFailedLoginException: ('Invalid authentication token.', PyiCloudAPIResponseException('Missing apple_id field (Missing apple_id field)'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/icloudpd/bin/icloud", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/cmdline.py", line 81, in main
    main_aux()
  File "/opt/icloudpd/lib/python3.12/site-packages/pyicloud_ipd/cmdline.py", line 336, in main_aux
    raise RuntimeError(message) from err
RuntimeError: Bad username or password for abcd@gmail.com
2024-08-03 16:18:12 ERROR    Keyring file does not exist. Please try again`

**Comment by oliveryu97** [comment]
me too

**Comment by golgo13a** [comment]
I get the same error... help!

**Comment by golgo13a** [comment]
I just figured it out... my bad! I didn't create a directory called /config in the docker folder.

**Comment by Austinzveare** [comment]
> I just figured it out... my bad! I didn't create a directory called /config in the docker folder.

Can you elaborate? In the Unraid template I have container path /config pointing to icloudpd's appdata folder, which creates an icloudpd.conf file, does it need to be in a folder titled config?

**Comment by Austinzveare** [comment]
> I just figured it out... my bad! I didn't create a directory called /config in the docker folder.

For anyone wondering this was indeed the case. I had config pointing to icloudpd's appdata folder, when it actually needed to point to a config folder within it's appdata folder.

**Comment by RezzZ** [comment]
> > I just figured it out... my bad! I didn't create a directory called /config in the docker folder.
> 
> For anyone wondering this was indeed the case. I had config pointing to icloudpd's appdata folder, when it actually needed to point to a config folder within it's appdata folder.

I'm not sure yet what to fix here. Running unraid too, but config is already mapped:
<img width="212" alt="UpdateContainer 2024-08-26 10-12-10" src="https://github.com/user-attachments/assets/3ad8e482-0903-4302-bd57-0c026250a617">
inside the container, going to /config gives me the icloudpd.conf file which has the required fields to login. What am I missing?

**Comment by RezzZ** [comment]
ok, I removed all variables from the unraid template and restarted the container. Initializing succeeded now :).

**Comment by Cantello** [comment]
> ok, I removed all variables from the unraid template and restarted the container. Initializing succeeded now :).

When you say *all* variables, do you really mean all of them? I am currently stuck on the `Bad username or password` as well, the regular script can login with the credentials, the docker container (via an Unraid template here) cannot.

**Comment by cfurrow** [comment]
It's possible that it's an upstream pyicloud issue

```
pyicloud_ipd.exceptions.PyiCloudAPIResponseException: Service Temporarily Unavailable (503)
```

https://github.com/picklepete/pyicloud/issues/456

> Authentication is now SRP6a. Old UID/pwd method gets the 503.

It could also be a digital certificate change:

https://github.com/gcobb321/icloud3_v3/issues/341
> Looks like the digital certificate for apple.com might have just been updated. It says Not valid before Saturday 19 October 2024 at 04:00:26 Australian Eastern Daylight Time, which was early this morning here.


I'm getting the "Service Temporarily Unavailable (503)" as well, and pretty sure the timing lines up with Oct 19 when it started to fail for me.

Will wait a few more days and check in on things. Could require a few more to fix depending on the complexity.

Pretty sure this is not a boredazfcuk/docker-icloudpd issue, or even a icloud-photos-downloader/icloud_photos_downloader issue.

**Comment by RezzZ** [comment]
> > ok, I removed all variables from the unraid template and restarted the container. Initializing succeeded now :).
> 
> When you say _all_ variables, do you really mean all of them? I am currently stuck on the `Bad username or password` as well, the regular script can login with the credentials, the docker container (via an Unraid template here) cannot.
> 
I only kept:
AppData Config Path
Synchronisation Interval (seconds)
Data
Time Zone
force_gid
(and the following under more settings:)
Command Line Options
Root
Directory Permissions
File Permissions
UMASK
User ID (PUID)
Group ID (PGID)

all other variables are stored in the 
ipcloudpd/config/icloudpd.conf file which is in your appdata folder

**Comment by boredazfcuk** [comment]
The only variable I create my container with is TZ=

The time zone is needed by the host OS, which is why I set it to be created with the container. Everything else is configured using the config file. The main reason for this is that all I do with the variables is write them to the config file, then use the config file to set the variables in the script.


> AppData Config Path
Not sure what this is. Unraid variable maybe?

> Synchronisation Interval (seconds)
Set via config file

> Data
Not sure what this is

> Time Zone
Not sure what this is, container uses TZ

> force_gid
Set via config file

> Command Line Options
This is no longer is used IIRC

> Root
Not sure what this is, Unraid again?

> Directory Permissions
Set via config file

> File Permissions
Set via config file

> UMASK
Not sure what this is, Unraid again maybe

> User ID (PUID)
Not sure what this is, Unraid again maybe

> Group ID (PGID)

Not sure what this is, Unraid again maybe

**Comment by boredazfcuk** [comment]
>pyicloud_ipd.exceptions.PyiCloudAPIResponseException: Service Temporarily Unavailable (503)

> Pretty sure this is not a boredazfcuk/docker-icloudpd issue, or even a icloud-photos-downloader/icloud_photos_downloader issue.

This

**Comment by boredazfcuk** [comment]
A proposed fix has submitted to the upstream project so I will update my container as soon as it's merged.

**Comment by RezzZ** [comment]
> The only variable I create my container with is TZ=
> 
> The time zone is needed by the host OS, which is why I set it to be created with the container. Everything else is configured using the config file. The main reason for this is that all I do with the variables is write them to the config file, then use the config file to set the variables in the script.
> 
> > AppData Config Path
> 
> Not sure what this is. Unraid variable maybe?
> 
> > Synchronisation Interval (seconds)
> 
> Set via config file
> 
> > Data
> 
> Not sure what this is
> 
> > Time Zone
> 
> Not sure what this is, container uses TZ
> 
> > force_gid
> 
> Set via config file
> 
> > Command Line Options
> 
> This is no longer is used IIRC
> 
> > Root
> 
> Not sure what this is, Unraid again?
> 
> > Directory Permissions
> 
> Set via config file
> 
> > File Permissions
> 
> Set via config file
> 
> > UMASK
> 
> Not sure what this is, Unraid again maybe
> 
> > User ID (PUID)
> 
> Not sure what this is, Unraid again maybe
> 
> > Group ID (PGID)
> 
> Not sure what this is, Unraid again maybe

yes, as I mentioned, all variables are from the unraid template file that uses this docker container.

**Comment by welshdylan** [comment]
has this been fixed now as i'm still getting the "bad username or password" or the 503 error?

**Comment by Symbiot78** [comment]
I've got the same issue. I've checked the pw so many times.
I set up 1 container with my icloud id.. works
then went to set up my wifes.. and am seeing the 'bad password....' 
I tried removing the first container just to see if there was some bleed-through.. but no.. same problem.
Repulled, recreated, restarted.. 
removed folders, recreated them.
removed the .conf file and let it recreate that then reconfigured it.

but nothing works.
